airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -833,6 +835,95 @@ definitions:
       $parameters:
         type: object
         additionalProperties: true
+  JwtAuthenticator:
+    title: JWT Authenticator
+    description: Authenticator for requests using JWT authentication flow.
+    type: object
+    required:
+      - type
+      - secret_key
+      - algorithm
+      - jwt_headers
+      - jwt_payload
+    properties:
+      type:
+        type: string
+        enum: [JwtAuthenticator]
+      secret_key:
+        type: string
+        description: Secret used to sign the JSON web token.
+        examples:
+          - "{{ config['secret_key'] }}"
+      algorithm:
+        type: string
+        description: Algorithm used to sign the JSON web token.
+        examples:
+          - "ES256"
+      token_duration:
+        type: integer
+        title: Token Duration
+        description: The amount of time in seconds a JWT token can be valid after being issued.
+        examples:
+          - 1200
```

Comment:
I haven't seen any consistency, although 1200s (20 min) is the smallest duration I've seen -- maybe a good default.

That being said, does this mean if a sync takes longer than 20 minutes it would fail? Should I include some sort of refresh mechanism? Or, is the authenticator re-instantiated per read?

Reply:
good question! can we refresh the token at runtime? [this is how we do it for oauth authenticators](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py#L58-L63)

Reply:
@girarda Actually I believe we're good -- each time the the request is prepared it will invoke _get_jwt_headers which will "refresh" the expiration time (and therefore refresh the token, as the token is the headers, payload, secret_key all encoded into a single string).